src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,156 @@
+                return Optional.of(shoppingCart);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to get cart with ID " + id, e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String selectAllShoppingCartsQuery = "SELECT * FROM carts;";
+        List<ShoppingCart> allShoppingCarts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(selectAllShoppingCartsQuery);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart shoppingCart = getShoppingCartFromResultSet(resultSet);
+                shoppingCart.setProducts(getProductsFromShoppingCartId(shoppingCart.getId()));
+                allShoppingCarts.add(shoppingCart);
+            }
+            return allShoppingCarts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to retrieve all carts", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart element) {
+        String updateShoppingCartQuery = "UPDATE carts SET user_id = ? "
+                + "WHERE card_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(updateShoppingCartQuery);
+            statement.setLong(1, element.getUserId());
+            statement.setLong(2, element.getId());
+            statement.executeUpdate();
+            deleteShoppingCartFromCartsProducts(element.getId());
+            insertCartsProducts(element);
+            return element;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to update " + element, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteShoppingCartQuery = "DELETE FROM carts WHERE card_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteShoppingCartFromCartsProducts(id);
+            PreparedStatement statement = connection.prepareStatement(deleteShoppingCartQuery);
+            statement.setLong(1, id);
+            int numberOfRowsDeleted = statement.executeUpdate();
+            return numberOfRowsDeleted != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to delete cart with ID " + id, e);
+        }
+    }
+
+    private void insertCartsProducts(ShoppingCart shoppingCart) throws SQLException {
+        String insertCartsProductsQuery = "INSERT INTO carts_products (cart_id, product_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : shoppingCart.getProducts()) {
+                PreparedStatement insertStatement =
```

Comment:
```suggestion
                var insertStatement = connection.prepareStatement(insertCartsProductsQuery);
```
can we use var here?